doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+                +--ro resource?           string
+                +--ro text?               string
+                +--ro time-created?       event:timeticks64
+                +--ro type-id?            string
+                +--ro severity?           event:severity-type
+                +--ro acknowledged?       boolean
+                +--ro acknowledge-time?   event:timeticks64
+```
+
+Following is for sonic yang to support event profiles.
+```
+module: sonic-evprofile
+
+  rpcs:
+    +---x get-evprofile
+    |  +--ro output
+    |     +--ro file-name?   string
+    |     +--ro file-list*   string
+    +---x set-evprofile
+       +---w input
+       |  +---w file-name?   string
+       +--ro output
+          +--ro status?   string
+```
+
+openconfig alarms yang is defined at [here](https://github.com/openconfig/public/blob/master/release/models/system/openconfig-alarms.yang)
+
+### 3.3.2 CLI
+#### 3.3.2.1 Exec Commands
+```
+sonic# alarm acknowledge <seq-id-of-raised-alarm>
+```
+An operator can acknolwedge a raised alarm. This indicates that the operator is aware of the fault condition and considers the condition not catastrophic. 
+Acknowledging an alarm updates alarm statistics and thereby applications like pmon can remove the particular alarm from status consideration.
+
+The alarm record in the ALARM table is marked with acknowledged field set to true. There is acknowledge-time field that indicates when that alarm is acknowledged.
+
+```
+sonic# alarm unacknowledge <seq-id-of-raised-alarm>
+```
+An operator can un-acknolwedge a previously acknowledged raised alarm. 
+Un-acknowledging an alarm updates alarm statistics and thereby applications like pmon can take the particular alarm into status consideration.
+
+The alarm record in the ALARM table is marked with acknowledged field set to false.
+There is acknowledge-time field that indicates when that alarm is un-acknowledged.
+
+```
+sonic# event profile <profile-name>
+```
+The command takes name of specified file, validates it for its syntax and values; merges it with its internal static map of events *static_event_map*.
+
+```
+sonic# clear event history
+```
+This command clears all the records in the event history table. All the event stats are cleared. 
+The command will not affect alarm table or alarm statistics.
+Eventd generates an event informing that event history table is cleared. 
+
+#### 3.3.2.2 Configuration Commands
+```
+sonic(config)# logging server <ip> [log|event] 
+```
+Note: The 'logging server' command is an existing, already supported command. 
+It is only enhanced to take either 'log' or 'event' to indicate either native syslog messages or syslog messages corresponding to events alone are sent to the remote host.
+Support with VRF/source-interface and configuring remote-port are all backward comaptible and will be applicable to either 'log' or 'event' options.
+
+#### 3.3.2.3 Show Commands
+```
+sonic# show event profile 
+--------------------------
```

Comment:
Why this output is considered as table? It is a list, right? In this case my suggestion for the output to be

Active Event Profile:
  myProfile.json

Available Event Profiles:
  default.json
  myProfile.json
  userProfile.json

Reply:
Your comment is about these that separates different data?
"--------------------------"

The o/p is displaying two parts: one is active profile and list of profiles.